library/src/blas3/Tensile/gemm.h

```diff
@@ -92,3 +92,251 @@ inline rocblas_status validateArgs(rocblas_handle    handle,
 
     return rocblas_status_success;
 } // validate parameters
+
+/*******************************************************************************
+ * temporary solution: complex gemm helper
+ * we will use tensile for final solution
+ ******************************************************************************/
+template <typename DT, typename ST>
+__global__ void kernel_copy_real_part_of_complex_matrix(DT*               dst,
+                                                        const rocblas_int ld_dst,
+                                                        const rocblas_int str_dst,
+                                                        const ST*         src,
+                                                        const rocblas_int ld_src,
+                                                        const rocblas_int str_src,
+                                                        const rocblas_int m,
+                                                        const rocblas_int n)
+{
+    size_t tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    size_t ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    dst = dst + hipBlockIdx_z * str_dst;
+    src = src + hipBlockIdx_z * str_src;
+
+    if(tx < m && ty < n)
+        dst[tx + ld_dst * ty] = src[tx + ld_src * ty].x;
+}
+
+template <typename DT, typename ST>
+__global__ void kernel_copy_real_image_of_complex_matrix(DT*               dst,
+                                                         const rocblas_int ld_dst,
```

Comment:
`imag` and `image` mean different things here. `copy_real_image` suggests a projection, while `copy_real_imag` suggests copying both `real` and `imag` parts. I suggest either `kernel_copy_complex_matrix` or `kernel_copy_real_part_of_complex_matrix` as names with clearer meanings.

Reply:
understand. thanks for catch that.